tests/flask/test_oauth2/test_client_configuration_endpoint.py

```diff
@@ -0,0 +1,505 @@
+from flask import json
+from authlib.common.security import generate_token
+from authlib.jose import jwt
+from authlib.oauth2.rfc7591.claims import ClientMetadataClaims
+from authlib.oauth2.rfc7592 import (
+    ClientConfigurationEndpoint as _ClientConfigurationEndpoint,
+)
+from tests.util import read_file_path
+from .models import db, User, Client, Token
+from .oauth2_server import TestCase
+from .oauth2_server import create_authorization_server
+
+
+class ClientConfigurationEndpoint(_ClientConfigurationEndpoint):
+    software_statement_alg_values_supported = ['RS256']
+
+    def authenticate_token(self, request):
+        auth_header = request.headers.get('Authorization')
+        if auth_header:
+            access_token = auth_header.split()[1]
+            return Token.query.filter_by(access_token=access_token).first()
+
+    def update_client(self, client, client_metadata, request):
+        client.set_client_metadata({**client.client_metadata, **client_metadata})
+        db.session.add(client)
+        db.session.commit()
+        return client
+
+    def authenticate_client(self, request):
+        client_id = request.uri.split('/')[-1]
+        return Client.query.filter_by(client_id=client_id).first()
+
+    def revoke_access_token(self, request, token):
+        token.revoked = True
```

Comment:
Do we need to add:

```
db.session.add(token)
db.session.commit()
```

Reply:
Tests were passing without this, but I suppose this is needed anyways.